Bearded.Graphics.SkiaSharp/SKBitmapTextureData.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.Runtime.InteropServices;
+using Bearded.Graphics.Textures;
+using OpenTK.Graphics.OpenGL;
+using SkiaSharp;
+
+namespace Bearded.Graphics.SkiaSharp;
+
+public sealed class SKBitmapTextureData : ITextureData
+{
+    private static readonly SKImageInfo info = SKImageInfo.Empty.WithColorType(SKColorType.Bgra8888);
+
+    private readonly SKBitmap image;
+
+    public int Width { get; }
+
+    public int Height { get; }
+
+    public static ITextureData From(string path) => new SKBitmapTextureData(SKBitmap.Decode(path, info));
+
+    public static ITextureData From(Stream stream) => new SKBitmapTextureData(SKBitmap.Decode(stream, info));
+
+    public static ITextureData From(SKBitmap bitmap) => new SKBitmapTextureData(bitmap);
+
+    public static ITextureData From(string path, IEnumerable<ITextureTransformation> transformations)
+    {
+        using var image = SKBitmap.Decode(path);
+        return From(image, transformations);
+    }
+
+    public static ITextureData From(Stream stream, IEnumerable<ITextureTransformation> transformations)
+    {
+        using var image = SKBitmap.Decode(stream);
+        return From(image, transformations);
+    }
+
+    public static ITextureData From(SKBitmap image, IEnumerable<ITextureTransformation> transformations)
+    {
+        if (image.ColorType != SKColorType.Bgra8888)
```

Comment:
The ImageSharp version of this method is more forgiving and does a copy internally. Should we pick a single approach and stick with it? My guess is that texture transformations are not agnostic of the pixel format either (because they act on the array directly), so perhaps just enforcing Bgra32 across the board is not such a bad idea. What do you think?

Reply:
I really don't like the pixel format aspect of the current interfaces, but I also don't know how to handle it best. Being opinionated is one way, at least it's consistent.

But realistically, why shouldn't we allow different formats? OpenGL accepts them happily - in fact, OpenGL textures support many different formats themselves, from different numbers of channels to different channel types (bytes, half floats, floats, doubles..). Especially with procedurally generated images, I see a lot of sense in generating 'special' textures - but even when we load things like normal and heightmaps that might have fewer channels, we could save memory by doing it more properly.

The downside is that this would make everything a lot more complicated, so I think it's better to handle that separately if desired.

Reply:
For now, I'm removing the exception and adding the same copying mechanism as in the ImageSharp version instead.